Framework/Algorithms/src/EditInstrumentGeometry.cpp

```diff
@@ -288,6 +283,12 @@ void EditInstrumentGeometry::exec() {
   instrument->markAsSource(source);
   source->setPos(0.0, 0.0, -1.0 * l1);
 
+  // Make a new bank, to add detectors to.
+  // Adding detectors to a bank, rather than directly to the root, allows this instrument
+  // to be successfully saved and reloaded by `NexusGeometrySave` and `NexusGeometryParser`.
+  // (Instrument will take ownership of the pointer.)
+  Geometry::CompAssembly *bank = new CompAssembly("Bank_1");
```

Comment:
I'm not very familiar with this, is there any chance this causes a name collision when the geometry is saved?

Reply:
There's no chance of that.  There is only one bank at all.  Previously `EditInstrumentGeometry` used no banks.  Basically, the instrument it creates is as _simple_ as possible.

Reply:
I'm realizing that I may not have answered your question.  This reminds me that I need to check whether or not the actual instrument names need to be unique.  (Mantid actually has an _instrument_ cache.  Like the ADS, but for instruments.)  Right now (, in the SNAPRed PR), the instruments get named as "SNAP_{grouping name}" -- I think probably they need (e.g.) part of the <state id>-hash as well.